### PR TITLE
Allow HF trainer to mask sequences prior to reduction

### PR DESCRIPTION
**Pull request by AAnoosheh**
### What does this PR do?

Type of change: ? 

Previously HF trainer did not account for loss masking

### Usage

```python
# Add a code snippet demonstrating how to use this
```

### Before your PR is "*Ready for review*"

Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/Model-Optimizer/blob/main/CONTRIBUTING.md) and your commits are signed (`git commit -s -S`).

Make sure you read and follow the [Security Best Practices](https://github.com/NVIDIA/Model-Optimizer/blob/main/SECURITY.md#security-coding-practices-for-contributors) (e.g. avoiding hardcoded `trust_remote_code=True`, `torch.load(..., weights_only=False)`, `pickle`, etc.).

- Is this change backward compatible?: ✅ / ❌ / N/A 
- If you copied code from any other sources or added a new PIP dependency, did you follow guidance in `CONTRIBUTING.md`: ✅ / ❌ / N/A 
- Did you write any new necessary tests?: ✅ / ❌ / N/A 
- Did you update [Changelog](https://github.com/NVIDIA/Model-Optimizer/blob/main/CHANGELOG.rst)?: ✅ / ❌ / N/A 

## Summary by CodeRabbit

* **Improvements**
  * Knowledge-distillation loss now properly ignores padding/special tokens and supports masked per-token averaging.
  * Default loss reduction behavior adjusted for finer-grained training control and clearer per-token outputs.
  * More robust logit handling with consistent numeric casting for improved stability and accuracy, including mixed-precision scenarios.

### CODE REVIEW

File: modelopt/torch/distill/plugins/huggingface.py [hunk 1]
```diff
@@ -15,12 +15,15 @@
...
 """ModelOpt plugin to train HuggingFace models with knowledge distillation."""
 
+from torch import Tensor
 from transformers.modeling_outputs import CausalLMOutputWithPast
 
 import modelopt.torch.distill as mtd
 from modelopt.torch.opt.plugins import ModelOptHFTrainer
 from modelopt.torch.utils import print_rank_0
 
+IGNORE_INDEX = -100
```
Comment (kevalmorabia97) [inline modelopt/torch/distill/plugins/huggingface.py]: what is this magic number?

Comment (AAnoosheh) [reply modelopt/torch/distill/plugins/huggingface.py]: Transformers padding label

Comment (kevalmorabia97) [reply modelopt/torch/distill/plugins/huggingface.py]: Lets use `nn.CrossEntropyLoss().ignore_index`
